src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -21,7 +23,7 @@
 
     private final AddressBook addressBook;
     private final UserPrefs userPrefs;
-    private final FilteredList<Person> filteredPersons;
+    private FilteredList<Person> filteredPersons;
```

Comment:
I think this instance of `FilterList<Person>` is final for a reason.

At start time of the application, `MainWindow.java` will invoke `LogicManager::getFilteredPersonList`, which invokes `ModelManager::getFilteredPersonList()`, which returns a instance of `FilteredList<Person>`, called `filteredPersons`

Any internal changes to `FilteredList<Person>` will trigger some event handler, which would cause the UI to update.
If you replace the `filteredPersons` here with a different instance, then `MainWindow` wont be tracking the same instance of `filteredPersons` anymore, thus there is no visual change to the list. 

I believe any other commands afterwards will also not cause any visual change anymore.

Reply:
However, I tried adding a person after searching -- the new person is displayed

Reply:
Hmm, not really sure what the actual behavior of `FilteredList` actually is.

From ModelManager, we tell the filteredPersons that is has been updated by invoking `FilteredList::setPredicate`. I assume this method will trigger some event listener in `FilteredList`. 

However, one possible explanation would be that the event listener in `FilteredList` will get triggered as long as the source list for `FilteredList` is changed, which is done when we add a new `Person` to `addressBook`

So, in our case, since MainWindow references the old `filteredPersons` instance, and this old instance still references `addressBook`, whenever a change is done to `addressBook`, this will trigger the eventListener in `filteredPersons` which updates the UI accordingly.

However, this is really relying on the assumption that `FilteredList` tracks the source list, and will update as long as the source list changes.

Reply:
This should really be `final`